### ISSUE TITLE
Hard coded code block in run directive 

### ISSUE THREAD

**Issue by projectoriented** [EichlerLab/pav#36]
Hello 👋 ,

Thank you for developing PAV! I'm seeing that this line is hard coded versus using the input directive, is this something you could update in the next release? If needed, I can make a PR.

https://github.com/EichlerLab/pav/blob/76c923265428331ab9a87e9db2e313acf0beaf74/rules/call.snakefile#L207

**Comment by paudano** [comment]
Pushed to the main branch, it will be in the next release.

It was an inconsistency, but shouldn't affect the pipeline. Was this causing you any trouble?

Thanks!

**Comment by projectoriented** [comment]
Hi again 👋 ,

I haven't got the chance to go through all of PAV yet (as a module in my pipeline) but I had to patch the hard coded reference + fai variables in a backwards compatible manner. Does it work for you to change them and I can submit a PR if you'd like. It would be great to not have any hard coded paths outside the context of snakemake rules because otherwise it becomes incompatible for when I import PAV as a workflow and force it to adhere to my paths.

Thanks in advance!

**Comment by paudano** [comment]
Those paths are coded as part of Snakemake rules, not the pav library. The reference is configured in the configuration JSON. Because PAV doesn't know what format the reference is in (i.e. bgzipped, gzipped, plain) or if it's indexed (fai), it creates a copy in the run directory in a format all rules can rely on, and that's the path you are referring to. The pipeline has to be pass files from one rule to another, and Snakemake does that as filename patterns. There's not really a way around that. I having a hard time understanding how this is affecting what you are trying to do.

**Comment by projectoriented** [comment]
Yup! It's not a problem from the pav library or how Snakemake creates the dag but the portability of global variables and hard-fixed paths within run directive when I use it as a module.

For example, here:
https://github.com/EichlerLab/pav/blob/ccd826a5f8c8b8982b183c1efb819ba9d9328430/rules/figures.snakefile#L183C49-L183C49

When I use PAV in my own workflow, I'm prefixing all outputs from PAV with a defined path. The example above with `REF_FAI` is not passed in as a definition within the directive blocks of the rule and therefore, it doesn't get its path  prefixed/modified. And then I get a FileNotFound error 😢 .

Will it be okay I open up a PR and we could go through multiple examples? Otherwise, I could just work off a forked version no problem!

**Comment by paudano** [comment]
I see. I am going to come back to this. I have some hard deadlines coming up, and I can look closer at this. There might be some simple things PAV can do to support your workflow.
